bin/crew

```diff
@@ -203,40 +215,66 @@ def list_installed
     @sorted_installed_packages.unshift('Package Version')
     @first_col_width = @sorted_installed_packages.map(&:split).map(&:first).max_by(&:size).size + 2
     @sorted_installed_packages.map(&:strip).each do |line|
-        puts "%-#{@first_col_width}s%s".lightgreen % line.split
+      puts "%-#{@first_col_width}s%s".lightgreen % line.split
     end
     puts
   end
 end
 
 def list_compatible(compat = true)
+  generate_compatible
   Dir[CREW_PACKAGES_PATH + '*.rb'].each do |filename|
     pkgName = File.basename filename, '.rb'
-    set_package pkgName, filename
-    if compat
-      if @pkg.compatibility.include? 'all' or @pkg.compatibility.include? ARCH
-        if File.exist? CREW_META_PATH + pkgName + '.filelist'
-          puts pkgName.lightgreen
-        else
-          puts pkgName
-        end
-      end
+    if pkgName.in? @device[:compatible_packages]
```

Comment:
Return to `if @pkg.compatibility.include? 'all' or @pkg.compatibility.include? ARCH`?

Reply:
The rationale is to just pull the compatibility list from device.json for the `list_compatible` function, otherwise you're just reading all the package files again and that's what `generate_compatible` is for.